Audible-Altimeter-Embedded/src/Tiny2350.cpp

```diff
@@ -0,0 +1,56 @@
+#include <Audible-Altimeter-Embedded/Tiny2350.hpp>
+#include <cstdint>
+
+#include "pico/stdlib.h"
+
+namespace {
+
+constexpr std::uint32_t LED_PIN_RED{18};
+constexpr std::uint32_t LED_PIN_GREEN{19};
+constexpr std::uint32_t LED_PIN_BLUE{20};
+
+auto is_initialized{false};
+
+auto initialize() {
+  if (is_initialized) {
+    return;
+  }
+
+  gpio_init(LED_PIN_RED);
+  gpio_init(LED_PIN_GREEN);
+  gpio_init(LED_PIN_BLUE);
+
+  gpio_set_dir(LED_PIN_RED, GPIO_OUT);
+  gpio_set_dir(LED_PIN_GREEN, GPIO_OUT);
+  gpio_set_dir(LED_PIN_BLUE, GPIO_OUT);
+
+  is_initialized = true;
+}
+
+}  // namespace
+
+namespace altimeter {
+
+Tiny2350::Tiny2350() { initialize(); }
+
+void Tiny2350::toggle_led(const Color color, const bool on) const {
```

Comment:
the function being const doesnt sit right with me because although its not mutating any class members, it is mutating resources the class manages i.e the gpio registers. There is user visible side effects to these functions.